### PR TITLE
Added flatEndedRigidIndenter contact case

### PR DESCRIPTION
**Pull request by iBatistic**

### CODE REVIEW

File: linearElasticity/curvedCantilever/coupledUnsLinearGeometryLinearElastic/Allrun [hunk 1]
```diff
@@ -7,7 +7,7 @@
 source solids4FoamScripts.sh
 
 # Compile functionObject for generation of analytical solution
-compileLibrary analyticalCurvedCantilever
+compileLibrary curvedCantileverAnalyticalSolution
```
Comment (philipcardiff) [inline linearElasticity/curvedCantilever/coupledUnsLinearGeometryLinearElastic/Allrun:10]: I did not know about the `compileLibrary` function; that is convenient. Unfortunately, it seems to only be in foam extend.

One option is that we could add this function to solids4FoamScripts.sh. For now, we don't need to do anything.

**Comment by philipcardiff** [comment]
@iBatistic, this looks great. It is now merged. Feel free to also push the case and README to the main solids4foam repo, and it can be added to the website.

Later, I expect we can add steps in the README to show how to use `solidSegmentContact` for this case (once it is added) and comment on its relative benefits.

**Comment by iBatistic** [comment]
Sure, agree!

File: linearElasticity/curvedCantilever/coupledUnsLinearGeometryLinearElastic/Allrun (hunk 1 above)
Comment (iBatistic) [reply linearElasticity/curvedCantilever/coupledUnsLinearGeometryLinearElastic/Allrun:10]: For now, I'm adding cases using `fe41`, and will check/modify them using other versions later. This function object is already in the development branch, so `solids4Foam::rcompileLibrary` is not needed; I guess that later we can easily add it if required.

**Pull request merged**